File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        
        int arr[26];
        bool ans = true;
        memset(arr, 0, sizeof(arr));
        
        for(int i = 0; i < magazine.length(); i++){
            arr[int(magazine[i]) - 97]++;
        }
        
        for(int i = 0; i < ransomNote.length(); i++){
            if (!arr[int(ransomNote[i]) - 97])
                return false;
            else
                arr[int(ransomNote[i]) - 97]--;
        }
        return true;

    }
};

File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        a, b = Counter(ransomNote), Counter(magazine)
        
        if len(b) < len(a):
            return False
    
        for k in a:
            if k not in b or a[k] > b[k]:
                return False
        return True